tests/integration/test_charm.py

```diff
@@ -1,34 +1,94 @@
 #!/usr/bin/env python3
-# Copyright 2024 Ubuntu
+# Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
 import asyncio
 import logging
 from pathlib import Path
 
+import juju
 import pytest
+import tenacity
 import yaml
+from helpers import bootstrap_microceph
 from pytest_operator.plugin import OpsTest
 
 logger = logging.getLogger(__name__)
 
-METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
-APP_NAME = METADATA["name"]
+BASES = ["ubuntu@22.04"]
+CLIENT = "cephfs-client"
+METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
+PROXY = METADATA["name"]
 
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest):
-    """Build the charm-under-test and deploy it together with related charms.
+@pytest.mark.skip_if_deployed
+@pytest.mark.parametrize("base", BASES)
+@pytest.mark.order(1)
+async def test_build_and_deploy(ops_test: OpsTest, base: str) -> None:
+    """Test that the CephFS server can stabilize against cephfs-client."""
+    (share_info, auth_info) = bootstrap_microceph()
```

Comment:
[nit]: You shouldn't need the parentheses here:

 ```suggestion
    share_info, auth_info = bootstrap_microceph()
```